relayer/message_relayer.go

```diff
@@ -51,48 +57,77 @@ var (
 // and send the signed warp message to the destination chain.
 // Each messageRelayer runs in its own goroutine.
 type messageRelayer struct {
-	relayer                 *Relayer
-	warpMessage             *avalancheWarp.UnsignedMessage
+	logger                  logging.Logger
+	metrics                 *MessageRelayerMetrics
+	network                 *peers.AppRequestNetwork
+	messageCreator          message.Creator
+	responseChan            chan message.InboundMessage
+	sourceBlockchain        config.SourceBlockchain
 	destinationBlockchainID ids.ID
+	originSenderAddress     common.Address
+	destinationAddress      common.Address
 	signingSubnetID         ids.ID
+	relayerKey              database.RelayerKey
 	warpQuorum              config.WarpQuorum
+	db                      database.RelayerDatabase
 }
 
 func newMessageRelayer(
-	relayer *Relayer,
-	warpMessage *avalancheWarp.UnsignedMessage,
-	destinationBlockchainID ids.ID,
+	logger logging.Logger,
+	metrics *MessageRelayerMetrics,
+	network *peers.AppRequestNetwork,
+	messageCreator message.Creator,
+	responseChan chan message.InboundMessage,
+	relayerKey database.RelayerKey,
+	db database.RelayerDatabase,
+	sourceBlockchain config.SourceBlockchain,
+	cfg *config.Config,
 ) (*messageRelayer, error) {
-	quorum, err := relayer.globalConfig.GetWarpQuorum(destinationBlockchainID)
+	quorum, err := cfg.GetWarpQuorum(relayerKey.DestinationBlockchainID)
 	if err != nil {
-		relayer.logger.Error(
+		logger.Error(
 			"Failed to get warp quorum from config. Relayer may not be configured to deliver to the destination chain.",
-			zap.String("destinationBlockchainID", destinationBlockchainID.String()),
+			zap.String("destinationBlockchainID", relayerKey.DestinationBlockchainID.String()),
 			zap.Error(err),
 		)
 		return nil, err
 	}
 	var signingSubnet ids.ID
-	if relayer.sourceSubnetID == constants.PrimaryNetworkID {
+	if sourceBlockchain.GetSubnetID() == constants.PrimaryNetworkID {
 		// If the message originates from the primary subnet, then we instead "self sign" the message using the validators of the destination subnet.
-		signingSubnet = relayer.globalConfig.GetSubnetID(destinationBlockchainID)
+		signingSubnet = cfg.GetSubnetID(relayerKey.DestinationBlockchainID)
 	} else {
 		// Otherwise, the source subnet signs the message.
-		signingSubnet = relayer.sourceSubnetID
+		signingSubnet = sourceBlockchain.GetSubnetID()
 	}
 	return &messageRelayer{
-		relayer:                 relayer,
-		warpMessage:             warpMessage,
-		destinationBlockchainID: destinationBlockchainID,
+		logger:                  logger,
+		metrics:                 metrics,
+		network:                 network,
+		messageCreator:          messageCreator,
+		responseChan:            responseChan,
+		sourceBlockchain:        sourceBlockchain,
+		destinationBlockchainID: relayerKey.DestinationBlockchainID,
+		originSenderAddress:     relayerKey.OriginSenderAddress,
+		destinationAddress:      relayerKey.DestinationAddress,
```

Comment:
Removed the address fields as they're not used anywhere. I left in `destinationBlockchainID` as a separate field since it's used extensively throughout the type